autosklearn/ensemble_builder.py

```diff
@@ -691,25 +697,81 @@ def predict(self, set_: str,
         # TODO: ADD saving of predictions on "ensemble data"
 
     def _delete_non_candidate_models(self, candidates):
-        candidates = [os.path.split(cand)[1] for cand in candidates]
-        for model_path in self.read_preds.keys():
-            if self.read_preds[model_path]['deleted']:
+        # Loop through the files currently in the directory
+        for pred_path in self.y_ens_files:
+
+            # Do not delete candidates
+            if pred_path in candidates:
                 continue
-            match = self.model_fn_re.search(model_path)
-            _num_run = int(match.group(2))
-            # Do not remove the dummy prediction!
-            if _num_run == 1:
+
+            match = self.model_fn_re.search(pred_path)
+            _full_name = match.group(0)
+            _seed = match.group(1)
+            _num_run = match.group(2)
+            _budget = match.group(3)
+
+            # Do not delete the dummy prediction
+            if int(_num_run) == 1:
                 continue
-            model_file = os.path.split(model_path)[1]
-            if model_file not in candidates:
+
+            # Besides the prediction, we have to take care of three other files: model,
+            # validation and test.
+            model_name = '%s.%s.%s.model' % (_seed, _num_run, _budget)
+            model_path = os.path.join(self.dir_models, model_name)
+            pred_valid_name = 'predictions_valid' + _full_name
```

Comment:
Hey, I believe these are not used anywhere, is that correct?

Reply:
No, actually they are. They go inside `locks`, a dictionary that hold the locks and the file paths of the files to be deleted.

Reply:
(Hummm, looking back, the data structure here doesn't need to be a dictionary, because I never access the values using the keys. Therefore I could have to separate lists, one for the locks and one for the file paths. I'm going to change this. I think it will increase the readability.)